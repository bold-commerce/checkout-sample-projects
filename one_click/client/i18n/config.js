import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;