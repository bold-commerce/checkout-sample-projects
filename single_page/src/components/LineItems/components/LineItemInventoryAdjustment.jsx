import React from 'react';
import { LineItemQuantity } from '.';
import ArrowRightIcon from './ArrowRightIcon';
import { useTranslation } from 'react-i18next';

const LineItemInventoryAdjustment = ({ originalQuantity, quantity, readOnly }) => {
  const { t } = useTranslation();
  if (quantity === 0) {
    return (
      <div className="LineItem__QuantityAdjustmentWrapper">
        <span className="LineItem__SoldOut">{t('line_item.sold_out')}</span>
      </div>
    );
  } else {
    return (
      <div className="LineItem__QuantityAdjustmentWrapper">
        <LineItemQuantity
          readOnly={readOnly}
          defaultValue={originalQuantity}
        />
        <ArrowRightIcon />
        <LineItemQuantity
          readOnly={readOnly}
          defaultValue={quantity}
        />
      </div>
    )
  }
};

export default LineItemInventoryAdjustment;
