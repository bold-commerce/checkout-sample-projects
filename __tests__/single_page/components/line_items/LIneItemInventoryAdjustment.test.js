import React from "react";
import { render } from "@testing-library/react";
import LineItemInventoryAdjustment from '../../../../single_page/src/components/LineItems/components/LineItemInventoryAdjustment';
import {
  exampleUseBillingSameAsShipping as MOCKbillingSameAsShipping,
  exampleUseBillingAddress as MOCKbillingAddress,
  exampleUsePaymentIframe as MOCKpaymentIframe,
  exampleUseCheckoutStore as MOCKcheckoutStore,
  exampleUseCountryInfo as MOCKcountryInfo,
  exampleUseLineItems as MOCKlineItems
} from '../../../utils/hookHelpers';
import '../../../../single_page/src/i18n/config';

jest.mock('@boldcommerce/checkout-react-components', () => ({
  ...jest.requireActual('@boldcommerce/checkout-react-components'),
  useBillingSameAsShipping: () => MOCKbillingSameAsShipping,
  useBillingAddress: () => MOCKbillingAddress,
  usePaymentIframe : () => MOCKpaymentIframe,
  useCheckoutStore: () => MOCKcheckoutStore,
  useCountryInfo: () => MOCKcountryInfo,
  useLineItems: () => MOCKlineItems
}))

describe('LineItemInventoryAdjustment', () => {
  test('renders LineItemInventoryAdjustment component', () => {
    const { asFragment } = render(
      <LineItemInventoryAdjustment
        originalQuantity={2}
        quantity={1}
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })

  test('renders LineItemInventoryAdjustment component sold out', () => {
    const { asFragment } = render(
      <LineItemInventoryAdjustment
        originalQuantity={2}
        quantity={0}
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })
  test('renders LineItemInventoryAdjustment component that is read only', () => {
    const { asFragment } = render(
      <LineItemInventoryAdjustment
        originalQuantity={2}
        quantity={1}
        readOnly
      />
    )

    expect(asFragment()).toMatchSnapshot();
  })
})
