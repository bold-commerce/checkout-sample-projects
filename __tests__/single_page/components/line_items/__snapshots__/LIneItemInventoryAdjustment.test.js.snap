// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineItemInventoryAdjustment renders LineItemInventoryAdjustment component 1`] = `
<DocumentFragment>
  <div
    class="LineItem__QuantityAdjustmentWrapper"
  >
    <div
      class="LineItem__LineItemQuantityWrapper"
    >
      <div
        class="LineItemQuantity__Decrement"
      >
        <button
          aria-busy="false"
          aria-disabled="false"
          class="stx-button"
        >
          <span
            class="stx-button__text"
          >
            -
          </span>
          <span
            class="stx-loading-spinner stx-button__spinner"
          />
        </button>
      </div>
      <div
        class="LineItemQuantity"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            type="number"
            value="2"
          />
        </div>
      </div>
      <div
        class="LineItemQuantity__Increment"
      >
        <button
          aria-busy="false"
          aria-disabled="false"
          class="stx-button"
        >
          <span
            class="stx-button__text"
          >
            +
          </span>
          <span
            class="stx-loading-spinner stx-button__spinner"
          />
        </button>
      </div>
    </div>
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.7239 7.99998L8.19526 2.47138C7.93491 2.21103 7.93491 1.78892 8.19526 1.52858C8.45561 1.26823 8.87772 1.26823 9.13807 1.52858L15.1381 7.52858C15.3984 7.78893 15.3984 8.21103 15.1381 8.47138L9.13807 14.4714C8.87772 14.7317 8.45561 14.7317 8.19526 14.4714C7.93491 14.211 7.93491 13.7889 8.19526 13.5286L13.7239 7.99998Z"
        fill="black"
        fill-opacity="0.65"
      />
      <path
        d="M13.3332 7.33331C13.7014 7.33331 13.9998 7.63179 13.9998 7.99998C13.9998 8.36817 13.7014 8.66665 13.3332 8.66665H1.33317C0.964981 8.66665 0.666504 8.36817 0.666504 7.99998C0.666504 7.63179 0.964981 7.33331 1.33317 7.33331L13.3332 7.33331Z"
        fill="black"
        fill-opacity="0.65"
      />
    </svg>
    <div
      class="LineItem__LineItemQuantityWrapper"
    >
      <div
        class="LineItemQuantity__Decrement"
      >
        <button
          aria-busy="false"
          aria-disabled="true"
          class="stx-button stx-button--is-disabled"
          disabled=""
        >
          <span
            class="stx-button__text"
          >
            -
          </span>
          <span
            class="stx-loading-spinner stx-button__spinner"
          />
        </button>
      </div>
      <div
        class="LineItemQuantity"
      >
        <div
          class="stx-input"
        >
          <input
            class="stx-input__input-element"
            type="number"
            value="1"
          />
        </div>
      </div>
      <div
        class="LineItemQuantity__Increment"
      >
        <button
          aria-busy="false"
          aria-disabled="false"
          class="stx-button"
        >
          <span
            class="stx-button__text"
          >
            +
          </span>
          <span
            class="stx-loading-spinner stx-button__spinner"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LineItemInventoryAdjustment renders LineItemInventoryAdjustment component sold out 1`] = `
<DocumentFragment>
  <div
    class="LineItem__QuantityAdjustmentWrapper"
  >
    <span
      class="LineItem__SoldOut"
    >
      Sold out
    </span>
  </div>
</DocumentFragment>
`;

exports[`LineItemInventoryAdjustment renders LineItemInventoryAdjustment component that is read only 1`] = `
<DocumentFragment>
  <div
    class="LineItem__QuantityAdjustmentWrapper"
  >
    <div
      class="LineItem__LineItemQuantityWrapper"
    >
      <div
        class="LineItemQuantity__Decrement"
      />
      <div
        class="LineItemQuantity LineItemQuantity--read-only"
      >
        <span
          class="LineItemQuantity__ReadOnlyValue"
        >
          2
        </span>
      </div>
      <div
        class="LineItemQuantity__Increment"
      />
    </div>
    <svg
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.7239 7.99998L8.19526 2.47138C7.93491 2.21103 7.93491 1.78892 8.19526 1.52858C8.45561 1.26823 8.87772 1.26823 9.13807 1.52858L15.1381 7.52858C15.3984 7.78893 15.3984 8.21103 15.1381 8.47138L9.13807 14.4714C8.87772 14.7317 8.45561 14.7317 8.19526 14.4714C7.93491 14.211 7.93491 13.7889 8.19526 13.5286L13.7239 7.99998Z"
        fill="black"
        fill-opacity="0.65"
      />
      <path
        d="M13.3332 7.33331C13.7014 7.33331 13.9998 7.63179 13.9998 7.99998C13.9998 8.36817 13.7014 8.66665 13.3332 8.66665H1.33317C0.964981 8.66665 0.666504 8.36817 0.666504 7.99998C0.666504 7.63179 0.964981 7.33331 1.33317 7.33331L13.3332 7.33331Z"
        fill="black"
        fill-opacity="0.65"
      />
    </svg>
    <div
      class="LineItem__LineItemQuantityWrapper"
    >
      <div
        class="LineItemQuantity__Decrement"
      />
      <div
        class="LineItemQuantity LineItemQuantity--read-only"
      >
        <span
          class="LineItemQuantity__ReadOnlyValue"
        >
          1
        </span>
      </div>
      <div
        class="LineItemQuantity__Increment"
      />
    </div>
  </div>
</DocumentFragment>
`;
